### PR TITLE
feat: edge gateway resource

### PR DESCRIPTION
**Pull request by eremid**
### Description of your changes

If you submit change in the provider code, please make sure to:

- [x] Write or modify examples in `examples/` directory
- [ ] Write or modify acceptance tests
- [x] Run `make generate` to ensure the doc was updated properly

### How has this code been tested

* terraform import
* terraform plan+apply
* terraform destroy

### CODE REVIEW

**Review by gaetanars — APPROVED** [review]

File: examples/resources/cloudavenue_edge_gateway/import.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+terraform import cloudavenue_edge_gateway.example 12345678-1234-1234-1234-123456789012
```
Comment (gaetanars) [inline examples/resources/cloudavenue_edge_gateway/import.sh:1]: Maybe you can specify in comment that the import is make with the `id`

File: examples/resources/cloudavenue_edge_gateway/resource.tf [hunk 2]
```diff
@@ -0,0 +1,3 @@
+resource "cloudavenue_edge_gateway" "example" {
+  vdc_name = "VDC_Frangipane"
```
Comment (gaetanars) [inline examples/resources/cloudavenue_edge_gateway/resource.tf:2]: You provide an example with `vdc_name` but I don't see this attribute in the schema.
The required attributes seems to be `owner_name` and `owner_type`.

**Review by gaetanars — CHANGES_REQUESTED** [review]

**Comment by gaetanars** [comment]
What is the comportment with the actual API that don't send `load balancing` state on Read?

File: internal/provider/edge_gateway_resource.go [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+		}
+
+		isNotFound = true
+	}
+
+	if isNotFound {
+		state = &edgeGatewaysResourceModel{
+			ID:       types.StringValue(""),
+			Timeouts: state.Timeouts,
+		}
```
Comment (gaetanars) [inline internal/provider/edge_gateway_resource.go:307]:
```suggestion
		resp.State.RemoveResource(ctx)

		return
```

File: internal/provider/edge_gateway_resource.go [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+		jobStatus, errGetJob := getJobStatus(auth, r.client, job.JobId)
+		if errGetJob != nil {
+			return nil, "", err
+		}
+
+		edgeID := ""
+
+		if jobStatus == "DONE" {
+			// get all edge gateways and find the one that matches the tier0_vrf_id and owner_name
+			gateways, _, errEdgesGet := r.client.EdgeGatewaysApi.ApiCustomersV20EdgesGet(r.client.auth)
```
Comment (gaetanars) [inline internal/provider/edge_gateway_resource.go:197]:
```suggestion
			gateways, _, errEdgesGet := r.client.EdgeGatewaysApi.ApiCustomersV20EdgesGet(auth)
```

File: internal/provider/edge_gateway_resource.go [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+	case "vdc":
+		job, httpR, err = r.client.EdgeGatewaysApi.ApiCustomersV20VdcsVdcNameEdgesPost(auth, body, plan.OwnerName.ValueString())
+		if apiErr := CheckApiError(err, httpR); apiErr != nil {
+			resp.Diagnostics.Append(apiErr.GetTerraformDiagnostic())
+			if resp.Diagnostics.HasError() {
+				return
+			}
+		}
+	case "vdc-group":
+		job, httpR, err = r.client.EdgeGatewaysApi.ApiCustomersV20VdcGroupsVdcGroupNameEdgesPost(r.client.auth, body, plan.OwnerName.ValueString())
```
Comment (gaetanars) [inline internal/provider/edge_gateway_resource.go:177]:
```suggestion
		job, httpR, err = r.client.EdgeGatewaysApi.ApiCustomersV20VdcGroupsVdcGroupNameEdgesPost(auth, body, plan.OwnerName.ValueString())
```

**Comment by azrod** [comment]
Please add AccTest

**Review by gaetanars — APPROVED** [review]

**Pull request merged**